moped-editor/src/views/projects/projectView/ProjectPhases.js

```diff
@@ -0,0 +1,582 @@
+          return updateProjectPhase({
+            variables: phase,
+          })
+            .then(() => {
+              // Refetch data
+              refetch();
+              projectViewRefetch();
+            })
+            .catch((err) => {
+              console.error(err);
+            });
+        }
+      });
+    }
+  };
+
+  /**
+   * Checks if phase being added or updated has a corresponding status and creates
+   * update object accordingly
+   * @param {string} mutationPhaseId - phase being added or updated in project phase table
+   * @returns {Object} Object that will be used in updates to project status
+   */
+  const getProjectStatusUpdateObject = (mutationPhaseId) => {
+    const newPhaseName = phaseNameLookup[mutationPhaseId];
+    const statusMapped = getStatusByPhaseName(newPhaseName);
+
+    return !!statusMapped
+      ? {
+          // There is a status with same name as phase
+          status_id: statusMapped.status_id,
+          current_status: statusMapped.status_name.toLowerCase(),
+          current_phase: newPhaseName.toLowerCase(),
+          current_phase_id: mutationPhaseId,
+        }
+      : {
+          // There isn't a status that matches the phase
+          status_id: 1,
+          current_status: "active",
+          current_phase: newPhaseName.toLowerCase(),
+          current_phase_id: mutationPhaseId,
+        };
+  };
+
+  /**
+   * DateFieldEditComponent - renders a Date type Calendar select
+   * @param {object} props - Values passed through Material Table `editComponent`
+   * @return {JSX.Element}
+   * @constructor
+   */
+  const DateFieldEditComponent = (props) => (
+    <TextField
+      name={props.name}
+      label={props.label}
+      type="date"
+      variant="standard"
+      value={props.value}
+      onChange={(e) => props.onChange(e.target.value)}
+      onKeyDown={(e) => handleKeyEvent(e)}
+      InputLabelProps={{
+        shrink: true,
+      }}
+    />
+  );
+
+  /**
+   * ToggleEditComponent - renders a toggle for True/False edit fields
+   * @param {object} props - Values passed through Material Table `editComponent`
+   * @return {JSX.Element}
+   * @constructor
+   */
+  const ToggleEditComponent = (props) => (
+    <Grid component="label" container alignItems="center" spacing={1}>
+      <Grid item>
+        <Switch
+          checked={props.value ?? false}
+          onChange={(e) => props.onChange(!props.value)}
+          color="primary"
+          name={props.name}
+          inputProps={{ "aria-label": "primary checkbox" }}
+          onKeyDown={(e) => handleKeyEvent(e)}
+        />
+      </Grid>
+    </Grid>
+  );
+
+  /**
+   * DropDownSelectComponent - Renders a drop down menu for MaterialTable
+   * @param {object} props - Values passed through Material Table `editComponent`
+   * @param {string} name - Field name
+   * @return {JSX.Element}
+   * @constructor
+   */
+  const DropDownSelectComponent = (props) => {
```

Comment:
I know this is pre-existing code, and this component is not used in Milestones, but it really should be extracted to a separate file.